pkg/infrastructure/clusterapi/clusterapi.go

```diff
@@ -448,25 +448,25 @@ type machineManifest struct {
 // extractIPAddress extracts the IP address from a machine manifest file in a
 // provider-agnostic way by reading only the "status" stanza, which should be
 // present in all providers.
-func extractIPAddress(manifestPath string) (string, error) {
+func extractIPAddress(manifestPath string) ([]string, error) {
 	data, err := os.ReadFile(manifestPath)
 	if err != nil {
-		return "", fmt.Errorf("failed to read machine manifest %s: %w", manifestPath, err)
+		return []string{}, fmt.Errorf("failed to read machine manifest %s: %w", manifestPath, err)
 	}
 	var manifest machineManifest
 	if err := yaml.Unmarshal(data, &manifest); err != nil {
-		return "", fmt.Errorf("failed to unmarshal manifest %s: %w", manifestPath, err)
+		return []string{}, fmt.Errorf("failed to unmarshal manifest %s: %w", manifestPath, err)
 	}
 
-	var ipAddr string
+	var ipAddr []string
 	for _, addr := range manifest.Status.Addresses {
 		switch addr.Type {
 		case clusterv1.MachineExternalIP:
-			ipAddr = addr.Address
+			ipAddr = append(ipAddr, addr.Address)
 		case clusterv1.MachineInternalIP:
 			// Prefer external IP when present
 			if len(ipAddr) == 0 {
```

Comment:
This check assumed there was only one `ipAddr`. With it being a multiple now, we should rethink this.

Reply:
One option is to prepend external IPs to the slice, so they are considered first.

Reply:
Another option is to sort `manifest.Status.Addresses` based on `Type` and return that.